From 9955b2dcf0b279abaee14847f44eb1d86fac173f Mon Sep 17 00:00:00 2001
From: Vijayanand Jitta <vjitta@codeaurora.org>
Date: Tue, 23 Feb 2021 12:35:35 +0530
Subject: [PATCH 31/35] ANDROID: dma-buf: Export is_dma_buf_file

Export is_dma_buf_file function which will be used
by several vendors to get dmabuf info.

Bug: 267248830
Bug: 264474028
Change-Id: Ic8f7dd4f0a620839ab19f52841e9a6541515133c
Signed-off-by: Vijayanand Jitta <vjitta@codeaurora.org>
Signed-off-by: T.J. Mercier <tjmercier@google.com>
(cherry picked from commit 409aa4d675770f28e3fec16977842796f7caab62)
---
 drivers/dma-buf/dma-buf.c | 5 ++---
 include/linux/dma-buf.h   | 1 +
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/dma-buf/dma-buf.c
+++ b/drivers/dma-buf/dma-buf.c
@@ -33,8 +33,6 @@
 
 #include "dma-buf-sysfs-stats.h"
 
-static inline int is_dma_buf_file(struct file *);
-
 struct dma_buf_list {
 	struct list_head head;
 	struct mutex lock;
@@ -552,10 +550,11 @@
 /*
  * is_dma_buf_file - Check if struct file* is associated with dma_buf
  */
-static inline int is_dma_buf_file(struct file *file)
+int is_dma_buf_file(struct file *file)
 {
 	return file->f_op == &dma_buf_fops;
 }
+EXPORT_SYMBOL_NS_GPL(is_dma_buf_file, DMA_BUF);
 
 static struct file *dma_buf_getfile(size_t size, int flags)
 {
--- a/include/linux/dma-buf.h
+++ b/include/linux/dma-buf.h
@@ -666,6 +666,7 @@
 	return !!attach->importer_ops;
 }
 
+int is_dma_buf_file(struct file *file);
 int dma_buf_get_each(int (*callback)(const struct dma_buf *dmabuf,
 		     void *private), void *private);
 struct dma_buf_attachment *dma_buf_attach(struct dma_buf *dmabuf,
